packages/liveblocks-yjs/src/awareness.ts

```diff
@@ -46,46 +48,93 @@ export class Awareness extends Observable<unknown> {
     this.doc = doc;
     this.room = room;
     this.clientID = doc.clientID;
+    // Add the clientId to presence so we can map it to connectionId later
+    this.room.updatePresence({
+      [Y_PRESENCE_KEY]: { clientID: this.clientID },
+    });
     this.othersUnsub = this.room.events.others.subscribe((event) => {
+      this.rebuildActorToClientMap(event.others);
       // When others are changed, we emit an event that contains arrays added/updated/removed.
       if (event.type === "leave") {
         // REMOVED
         this.emit("change", [
-          { added: [], updated: [], removed: [event.user.connectionId] },
+          {
+            added: [],
+            updated: [],
+            removed: [this.actorToClientMap.get(event.user.connectionId)],
```

Comment:
Looks like this `this.actorToClientMap.get(event.user.connectionId)` part is the same everywhere below. Could we put this in a variable? Would also help if these blocks were oneliners again for readability.